snuba/clickhouse/query_inspector.py

```diff
@@ -38,6 +88,16 @@ def _visit_join(self, data_source: JoinClause[Table]) -> None:
         self.visit_join_clause(data_source)
 
     def _visit_simple_query(self, data_source: ProcessableQuery[Table]) -> None:
+        time_range = _get_date_range(data_source)
+        if time_range and (
+            self.__max_time_range is None or time_range > self.__max_time_range
+        ):
+            self.__max_time_range = time_range
+
+        self.__has_complex_conditions = (
```

Comment:
Should this only happen if `__max_time_range` is updated?

Reply:
I don't think so. Complex condition and max time range are independent